src/app/shared/base-value-accessor/base-value-accessor.directive.ts

```diff
@@ -0,0 +1,43 @@
+import { AfterViewInit, Directive, Input, ViewChild } from '@angular/core';
+import { ControlValueAccessor, DefaultValueAccessor } from '@angular/forms';
+import { ReplaySubject } from 'rxjs';
+
+@Directive({
+  selector: 'base-value-accessor',
+})
+export class BaseValueAccessorDirective
+  implements ControlValueAccessor, AfterViewInit {
+  @Input() formControlName: string;
+  @ViewChild(DefaultValueAccessor) valueAccessor: DefaultValueAccessor;
+
+  protected delegatedMethodCalls = new ReplaySubject<
+    (_: ControlValueAccessor) => void
+  >();
+
+  ngAfterViewInit(): void {
+    this.delegatedMethodCalls.subscribe((fn) => fn(this.valueAccessor));
```

Comment:
And I don't understand why ReplaySubject is used here.
I see you want to create a base class. Okay, you're able to do something like 

export abstract class BaseControl implements ControlValueAccessor